File: src/main/java/net/xalcon/technomage/common/blocks/BlockTMTileProvider.java
package net.xalcon.technomage.common.blocks;

import net.minecraft.block.ITileEntityProvider;
import net.minecraft.block.material.Material;
import net.minecraft.tileentity.TileEntity;

import javax.annotation.Nullable;

public abstract class BlockTMTileProvider extends BlockTM implements ITileEntityProvider
{
    public BlockTMTileProvider(String internalName, Material materialIn)
    {
        super(internalName, materialIn);
    }

    @Nullable
    public abstract Class<? extends TileEntity> getTileEntityClass();
}

File: src/main/kotlin/net/xalcon/technomage/common/blocks/BlockTMTileProvider.kt
package net.xalcon.technomage.common.blocks

import net.minecraft.block.ITileEntityProvider
import net.minecraft.block.material.Material
import net.minecraft.tileentity.TileEntity

abstract class BlockTMTileProvider(internalName: String, material: Material) :
        BlockTM(internalName, material), ITileEntityProvider
{
    abstract fun getTileEntityClasses():Iterable<Pair<String, Class<out TileEntity>>>
}